packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -257,7 +318,36 @@ async function processBlogSourceFile(
     return undefined;
   }
 
-  const tagsBasePath = normalizeUrl([baseUrl, options.routeBasePath, 'tags']); // make this configurable?
+  const tagsBasePath = normalizeUrl([baseUrl, options.routeBasePath, 'tags']);
+  let author: Author = {};
+
+  if (frontMatter.author_key) {
+    if (!authorList) {
+      throw Error(
+        `The "author_key" front matter is used but no author list file is found at path ${options.authorListPath}.`,
+      );
+    }
+    if (!authorList[frontMatter.author_key]) {
+      throw Error(
+        `Author with key ${frontMatter.author_key} not found in the list file.`,
+      );
+    }
+    author = authorList[frontMatter.author_key];
```

Comment:
ensure all authors are always found
display list of available keys in error message for conveniency